src/event/sdam.rs

```diff
@@ -0,0 +1,206 @@
+//! Contains the events and functionality for monitoring Server Discovery and Monitoring.
+
+use std::time::Duration;
+
+use serde::Deserialize;
+
+use crate::{
+    bson::{oid::ObjectId, Document},
+    error::Error,
+    options::ServerAddress,
+};
+
+pub use crate::sdam::{
+    public::{ServerInfo as ServerDescription, TopologyInfo as TopologyDescription},
```

Comment:
If we use `type` here instead of `use`, we can elide the lifetime too:
e.g.
```rust
pub type ServerDescription = ServerInfo<'static>;
```
I think from the user's perspective `pub use` and `pub type` are otherwise pretty much the same thing.